### DISCUSSION TITLE
pyright can't resolve PyQt (PySide6) events?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/pyright#3336] [Q&A]
Hello I have a stupid problem with **pyright** I can't solve myself.

I have the following class file which works without any problems:

```
from random import choice
from PySide6.QtWidgets import QPushButton, QMainWindow, QLabel, QLineEdit, QVBoxLayout, QWidget

window_titles = [
    "Some sample text",
    "Even more text",
    "Oh no! No more lemmings!",
    "This is really a lot of text",
    "Stop texting me"
]

class MainWindow(QMainWindow):

    def __init__(self):
        super().__init__()

        self.label = QLabel()

        self.input = QLineEdit()
        self.input.textChanged.connect(self.label.setText)

        self.button = QPushButton("Press Me!")
        self.button.clicked.connect(self.clicke_di_click)

        layout = QVBoxLayout()
        layout.addWidget(self.input)
        layout.addWidget(self.label)
        layout.addWidget(self.button)

        container = QWidget()
        container.setLayout(layout)

        self.setMinimumSize(300, 240)
        self.setMaximumSize(600, 480)
        self.setWindowTitle("My App")

        self.windowTitleChanged.connect(self.window_title_change)

        self.setCentralWidget(container)

    def clicke_di_click(self):
        new_window_title = choice(window_titles)
        self.button_toggle_status = self.button.isChecked()
        self.setWindowTitle(new_window_title)

    def window_title_change(self, window_title):
        print(f"Oh no, the window title changed to: {window_title}")

    def __str__(self):
        return "Window Module"
```

But when checking my code with **pyright** on the command line (installed with pip) I get the following output when linting this code:

> No configuration file found.
> pyproject.toml file found at /home/username/coding/qt-playground/pyside_basics/pyside_basics.
> Loading pyproject.toml file at /home/username/coding/qt-playground/pyside_basics/pyside_basics/pyproject.toml
> Assuming Python platform Linux
> Searching for source files
> Found 1 source file
> /home/username/coding/qt-playground/pyside_basics/pyside_basics/pytest/Window.py
>   /home/username/coding/qt-playground/pyside_basics/pyside_basics/pytest/Window.py:21:20 - error: Cannot access member "textChanged" for type "QLineEdit"
>     Member "textChanged" is unknown (reportGeneralTypeIssues)
>   /home/username/coding/qt-playground/pyside_basics/pyside_basics/pytest/Window.py:24:21 - error: Cannot access member "clicked" for type "QPushButton"
>     Member "clicked" is unknown (reportGeneralTypeIssues)
>   /home/username/coding/qt-playground/pyside_basics/pyside_basics/pytest/Window.py:38:14 - error: Cannot access member "windowTitleChanged" for type "MainWindow"
>     Member "windowTitleChanged" is unknown (reportGeneralTypeIssues)
> 3 errors, 0 warnings, 0 informations 
> Completed in 2.019sec

I can ignore it but I would really like to use **coc-pyright** inside **vim** and this addon shows me also three errors in the editor.

What am I doing wrong and how can I solve it?

**Comment by erictraut** [comment]
It looks like you're using a "pyproject.toml" configuration file. What configuration options have you provided for pyright?

Which version of PySide6 do you have installed? The latest version is a "py.typed" library, which means that it includes inlined type annotations and should therefore work with static type checkers. When I ran "pip install PySide6 --upgrade", it installed version 6.2.4. When I use this version with pyright's default settings, I'm not able to repro the errors you're seeing. That means your configuration must be different than mine.

**Reply by LaBatata101** [reply]
I'm having the same issue here using PySide6 version 6.3.0
Code:
```python
import sys
from PySide6 import QtCore, QtWidgets, QtGui


class HelloWorld(QtWidgets.QWidget):
    def __init__(self):
        super().__init__()

        self.button = QtWidgets.QPushButton("Push me!")
        self.button.clicked.connect(self.magic)

    @QtCore.Slot()
    def magic(self):
        print("magic")


def run():
    app = QtWidgets.QApplication()

    widget = HelloWorld()
    widget.setWindowTitle("Hello World!")
    widget.resize(500, 500)
    widget.show()

    sys.exit(app.exec())
```

Output:
```bash
No configuration file found.
pyproject.toml file found at /home/labatata/Code/Python/tim-gui.
Loading pyproject.toml file at /home/labatata/Code/Python/tim-gui/pyproject.toml
Pyproject file "/home/labatata/Code/Python/tim-gui/pyproject.toml" is missing "[tool.pyright]" section.
stubPath /home/labatata/Code/Python/tim-gui/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/home/labatata/Code/Python/tim-gui/tim_gui/gui/__init__.py
  /home/labatata/Code/Python/tim-gui/tim_gui/gui/__init__.py:16:21 - error: Cannot access member "clicked" for type
"QPushButton"
    Member "clicked" is unknown (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
Completed in 2.1sec
```

**Reply by erictraut** [reply]
The class `QPushButton` defined in the file `PySide6/QtWidgets.pyi` does not contain a method or attribute called `clicked`, nor do any of its parent classes. If you think that the stub file is in error and this class should expose a member with that name, please file a bug with the maintainers of PySide6.

**Reply by LaBatata101** [reply]
Thanks for the quick response!
